include/mbedtls/aes.h

```diff
@@ -127,26 +130,30 @@ int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
 
 #if defined(MBEDTLS_CIPHER_MODE_CBC)
 /**
- * \brief          AES-CBC buffer encryption/decryption
- *                 Length should be a multiple of the block
- *                 size (16 bytes)
- *
- * \note           Upon exit, the content of the IV is updated so that you can
- *                 call the function same function again on the following
- *                 block(s) of data and get the same result as if it was
- *                 encrypted in one call. This allows a "streaming" usage.
- *                 If on the other hand you need to retain the contents of the
- *                 IV, you should either save it manually or use the cipher
- *                 module instead.
- *
- * \param ctx      AES context
- * \param mode     MBEDTLS_AES_ENCRYPT or MBEDTLS_AES_DECRYPT
- * \param length   length of the input data
- * \param iv       initialization vector (updated after use)
- * \param input    buffer holding the input data
- * \param output   buffer holding the output data
- *
- * \return         0 if successful, or MBEDTLS_ERR_AES_INVALID_INPUT_LENGTH
+ * \brief This function performs the operation defined in the \p mode parameter (encrypt/decrypt), on the AES-CBC input data buffer defined in the \p ctx parameter. 
+ * It can be called as many times as needed, until all the input data is processed.\par
+ * mbedtls_aes_init(),  mbedtls_aes_setkey_enc() or mbedtls_aes_setkey_dec() must be called before 
+ * the first call to this API with the same context.
+ * The function processes the last data block if needed, finalizes the AES-CBC operation, 
+ * and produces operation results for MAC operations. 
+ *
+ * \note Upon exit, the content of the IV is updated so that you can
+ *       call the same function again on the next
+ *       block(s) of data and get the same result as if it was
+ *       encrypted in one call. This allows a "streaming" usage.
+ *       If you need to retain the contents of the IV, you should 
+ *       either save it manually or use the cipher module instead.
+ *
+ * \note The output buffer may be NULL for MAC operations.
+ *
+ * \param ctx      The AES context to encrypt/decrypt.
+ * \param mode     The AES mode: #MBEDTLS_AES_ENCRYPT or #MBEDTLS_AES_DECRYPT.
+ * \param length   The length of the input data in Bytes. AES-CBC buffer encryption/decryption length must be a multiple of the block size (16 Bytes).
+ * \param iv       Initialization vector (updated after use).
+ * \param input    The buffer holding the input data.
+ * \param output   The buffer holding the output data.
+ *
+ * \return         Zero if successful or error.h for the specific error code on failure.
```

Comment:
Same comment as before. We don't return error.h. Need actual error codes here

Reply:
fixed